velox/connectors/hive/storage_adapters/abfs/RegisterAbfsFileSystem.cpp

```diff
@@ -25,23 +25,25 @@ namespace facebook::velox::filesystems::abfs {
 #ifdef VELOX_ENABLE_ABFS
 folly::once_flag abfsInitiationFlag;
 
-static std::function<std::shared_ptr<FileSystem>(
-    std::shared_ptr<const Config>,
-    std::string_view)>
-    filesystemGenerator = [](std::shared_ptr<const Config> properties,
-                             std::string_view filePath) {
-      static std::shared_ptr<FileSystem> filesystem;
-      folly::call_once(abfsInitiationFlag, [&properties]() {
-        filesystem = std::make_shared<AbfsFileSystem>(properties);
-      });
-      return filesystem;
-    };
+std::function<std::shared_ptr<
+    FileSystem>(std::shared_ptr<const Config>, std::string_view)>
+abfsFileSystemGenerator() {
```

Comment:
I fixed this for S3 here https://github.com/facebookincubator/velox/pull/7388/files
But this can simply be a function 
```
std::shared_ptr<FileSystem> abfsFileSystemGenerator(
    std::shared_ptr<const Config> properties,
    std::string_view filePath) {}
```
and we can use `std::function(abfsFileSystemGenerator)` in the registry.

Reply:
hey @majetideepak  super thanks for quick review.

Your PR i think is mainly address multiple instance support, this PR used to make sure abfs file system register function been called during hive connector registeration to make sure we have abfs filesystem instance avaliable. One is ok right now for unblocking our prod end to end.


I would send another PR to address multiple instance support once your S3 PR merged (I can reference.)

does that ok for you ?

Reply:
@gaoyangxiaozhu My comment was on the `abfsFileSystemGenerator` API and not supporting multiple instances. Sorry if that was not clear. The current API of abfsFileSystemGenerator returns a function pointer which is not necessary. We can simplify it to a simple function returning a FileSystem instance.

Reply:
oh, i see, thank you @majetideepak for clarification, let me try refactor.

Reply:
done. please help review @majetideepak , thank you!